### ISSUE TITLE
Reimplement dark mode

### ISSUE THREAD

**Issue by kalicki1** [privacy-tech-lab/gpc-optmeowt#183]
Since issue #182 involved deleting our existing libraries, we will need to reimplement dark mode for OptMeowt at some point in the future. This is a low priority issue, we can get to it when we have time. 

For reference, I think we can reuse the same style sheets we have saved in an older commit prior to deleting our libraries, potentially with a better or more popular dark mode node package.

**Comment by SebastianZimmeck** [comment]
There might be an issue with the dependency. Maybe, it is better to use a different library that works with node.

**Comment by SebastianZimmeck** [comment]
Here is a link to a previous commit where OptMeowt has dark mode:

https://github.com/privacy-tech-lab/optmeowt-browser-extension/tree/3ce729d0688d74e558f646a9f28b0be8ce00d917

**Comment by SebastianZimmeck** [comment]
Maybe a good first issue for @OliverWang13 as the re-implementation of the dark mode may be fairly independent of analysis functionality or other more complex functionality.

**Comment by kalicki1** [comment]
Sounds good! I'm actually taking a cursory look at this right now, I have a list of candidate libraries. These are the "most popular" libs I could find. 

[dark-mode-switch](https://www.npmjs.com/package/dark-mode-switch): The old code uses this. Not as popular as the two below. Basically a bootstrap button.
[darkmode-js](https://www.npmjs.com/package/darkmode-js): This may be the way to go, looks very easy to implement vs. the library above. You'll just need to play with the old style sheets most likely. I think it creates a Darkmode object w/ storage implemented and built-in methods.
[darkmodejs](https://www.npmjs.com/package/@assortment/darkmodejs): I think this is an API for pulling your local machine's dark mode options if we ever want that. We did not have this before.

Going to leave this here for reference. Old style sheets are in the commit mentioned by @SebastianZimmeck.

**Comment by SebastianZimmeck** [comment]
Great, @kalicki1! [That](https://github.com/privacy-tech-lab/optmeowt-browser-extension/issues/183#issuecomment-884650902) will be a good starting point for @OliverWang13 (possibly with a little help from friends, i.e., you :).

**Comment by kalicki1** [comment]
@SebastianZimmeck, I have thought about this and think it might be best to go ahead with #197 without the dark mode. Once @OliverWang13 has had some time to get familiar with the project, we can quietly push an update to the Chrome and FF stores after we work through the reimplementation. This might be a little easier than trying to rush through it.

**Comment by OliverWang13** [comment]
@kalicki1 and I used the darkmode-js plugin to implement a dark mode. We used messages to allow communication between the settings page and the popup. We tried to use message passing to create a dark mode switch on the settings page, but we suspect that messages can not be properly sent between main-view and settings. The next step is working on the dark-mode.css files to update the views to our liking.

**Comment by SebastianZimmeck** [comment]
> but we suspect that messages can not be properly sent between main-view and settings

Does that mean that the dark mode would only work for the popup? Or would it be that the dark mode switch on the popup also creates dark mode on the settings pages?

**Comment by OliverWang13** [comment]
Currently, dark mode is working where if you click the switch on the popup, it changes for the entire site. We thought about adding another dark mode switch on the settings page but it was not functioning properly.

**Comment by SebastianZimmeck** [comment]
OK, I think that is totally sufficient. Good work, @OliverWang13 and @kalicki1! Feel free to close this issue once the change is in the main branch.

**Comment by kalicki1** [comment]
@OliverWang13 and I have been working on this issue on and off for a few days now and have made some progress.

I just pushed some changes that rolled back a bit of the code from the last commit and reimplemented a modified version of the `darkmode-js` library. The old commit had successfully synced the dark mode behavior between the popup and the options page, but not the styles.

I removed all of the extraneous code from the dark mode library that implemented extra buttons and a widget, etc., so that the code only manages one CSS class attribute, `darkmode--activated`.  I added this new Darkmode class to `src/theme/darkmode.js`.  

Technical note: I couldn't figure out why removing everything but the storage and the CSS broke the sync between the popup and the options page, but it seems that checking if the window is a browser via `const IS_BROWSER = typeof window !== 'undefined';` is the key that makes it work.

I think we will be able to test this and merge it back into main tomorrow.
